### ISSUE TITLE
Serialization don't support nullable array type

### ISSUE THREAD

**Issue by roquie** [temporalio/sdk-php#246]
### What are you really trying to do?

Pass a DTO object as argument to method in activity/workflow.

### Describe the bug

When developer using class as DTO with public properties to pass values to workflow with property type `?array` it got an error:

```
InvalidArgumentException: Passed value must be a type of array, but null given in /backend/vendor/temporal/sdk/src/Internal/Marshaller/Type/ArrayType.php:64\nStack trace:\n#0 /backend/vendor/temporal/sdk/src/Internal/Marshaller/Mapper/AttributeMapper.php(195): Temporal\\Internal\\Marshaller\\Type\\ArrayType->parse(NULL, NULL)
```

### Minimal Reproduction

1. Create a DTO class like this: 
```php
class DTO {
  public function __construct(
     public readonly ?array $foos
  ) {}
}
```
2. Create a workflow class with method `handle` and agruments like `handle(DTO $test)` 
3. Run

### Environment/Versions

- OS and processor: M1 Mac and Mac without M1
- Temporal Version: latest `1.*` and latest `2.*`
- Are you using Docker or Kubernetes or building Temporal from source?: Docker

**Comment by roquie** [comment]
https://github.com/temporalio/sdk-php/pull/245

**Comment by rustatian** [comment]
Closed since #245 was merged.
